src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1465,7 +1466,7 @@ void TorrentImpl::applyFirstLastPiecePriority(const bool enabled)
 
     // Download first and last pieces first for every file in the torrent
 
-    std::vector<lt::download_priority_t> piecePriorities = nativeHandle().get_piece_priorities();
+    auto piecePriorities = std::vector<lt::download_priority_t>(m_torrentInfo.piecesCount(), LT::toNative(DownloadPriority::Ignored));
```

Comment:
I've always felt this constructor syntax (looking like function declaration) as confusing. So if you don't insist too much, I wouldn't change it.

Reply:
This is about readability/uniformity. Are you ok with liberal use of this specific syntax? and more importantly, mixing it with the old constructor syntax without restrictions?

Reply:
> This is about readability/uniformity.

It looks readable for me (just like `auto name = QString::fromUtf8(...);`). At least it looks much better than `Type name(...);`.


> Are you ok with liberal use of this specific syntax?

IMO, it should be used in case when the only alternative is `Type name(...);`.

Reply:
>IMO, it should be used in case when the only alternative is Type name(...);.

OK, let's give it a try.